everywhere: reindent `.overlay` files with tabs

JordanYates opened the pull request:
`checkpatch.pl` requires that dts sources are indented with tabs, fix all the spaces that slipped in while checkpatch wasn't watching (#74570).

jeppenodgaard commented on boards/shields/esp_8266/boards/numaker_pfm_m467.overlay:
```diff
@@ -5,14 +5,14 @@
 &pinctrl {
-    uart2_esp8266: uart2_esp8266 {
+	uart2_esp8266: uart2_esp8266 {
 	group0 {
```
Needs indentation.

jeppenodgaard commented on samples/drivers/spi_flash/boards/stm32h573i_dk.overlay:
```diff
@@ -5,9 +5,9 @@
+	/* request 57 for XSPI1 */
 	dmas = <&gpdma1 4 57 STM32_DMA_PERIPH_TX
-		&gpdma1 5 57 STM32_DMA_PERIPH_RX>;
+			&gpdma1 5 57 STM32_DMA_PERIPH_RX>;
```
Is tabs and then a few spaces allowed to align entries? If so, this and other similar changes should probably remain unchanged.

JordanYates replied:
I was performing the alignment based on 4 character tabs, which I now see is not the default for dts source. I will do another parse over the files

jeppenodgaard reviewed (commented):
Nice to get this fixed!

JordanYates committed:
everywhere: reindent `.overlay` files with tabs

`checkpatch.pl` requires that dts sources are indented with tabs,
fix all the spaces that slipped in while checkpatch wasn't watching.

Signed-off-by: Jordan Yates <jordan@embeint.com>

henrikbrixandersen commented:
Please rebase on `main` and resolve any merge conflicts.

JordanYates commented:
> Please rebase on `main` and resolve any merge conflicts.

done

aescolar commented:
@JordanYates Github demands a rebase for some reason..

JordanYates commented:
> @JordanYates Github demands a rebase for some reason.. (and your original branch is protected so I can't push the rebase directly to it)

Odd indeed, no conflicts appeared while rebasing...
Done

fabiobaltieri reviewed (approved):
thanks!
